classes/services/SubmissionService.inc.php

```diff
@@ -32,6 +33,52 @@ public function __construct() {
 		\HookRegistry::register('Submission::getProperties::values', array($this, 'modifyPropertyValues'));
 	}
 
+	/**
+	 * Modify the isPublic check on a submission, based on whether it is scheduled
+	 * for publication in an issue and that issue is published.
+	 *
+	 * @param $hookName string
+	 * @param $args array [
+	 *		@option boolean Is it public?
+	 *		@option Submission
+ 	 * ]
+	 */
+	public function modifyIsPublic($hookName, $args) {
+		$isPublic = &$args[0];
+		$submission = $args[1];
+
+		if (is_a($submission, 'PublishedArticle')) {
+			$publishedArticle = $submission;
+		} else {
+			$publishedArticleDao = \DAORegistry::getDAO('PublishedArticleDAO');
+			$publishedArticle = $publishedArticleDao->getPublishedArticleByBestArticleId(
+				$submission->getContextId(),
+				$submission->getId(),
+				true
+			);
+		}
+
+		if (empty($publishedArticle)) {
+			$isPublic = false;
```

Comment:
Since the default presumption is `false`, and the goal of a hook structure here is to allow an ecosystem of rules to interoperate, I'd suggest only altering `$isPublic` when you want to vary from the default presumption (= grant access). Otherwise we'll probably have to work a lot with the order of hook registrants in encouraging their cooperation, which could be messier than  necessary.

Reply:
Oh good thinking! :+1: